src/expressions/projectionExpressionChecker.js

```diff
@@ -1,39 +1,135 @@
 import { BaseSubExpressionChecker } from './baseSubExpressionChecker'
+import { wrapWithDefault } from '../defaultFactResolver'
 
 export class ProjectionExpressionChecker extends BaseSubExpressionChecker {
+  /**
+   * Create a SubExpressionChecker
+   * @param {ServiceProvider} serviceProvider
+   */
+  constructor (serviceProvider) {
+    super(serviceProvider)
+    this.scopeCheckers = {
+      'single': new SingleScopeProjectionExpressionChecker(serviceProvider),
+      'all': new AllScopeProjectionExpressionChecker(serviceProvider),
+      'some': new SomeScopeProjectionExpressionChecker(serviceProvider)
+    }
+  }
+
   async checkSubExpression (fact, ssid, context) {
-    const core = this._getFactChecker().getAbundanceService().getCoreAPI()
-    const lawregContext = context
+    const scope = fact.scope ? fact.scope : 'single'
+    return this.scopeCheckers[scope].checkSubExpression(fact, ssid, context)
+  }
+}
 
+class BaseScopeProjectionExpressionChecker extends BaseSubExpressionChecker {
+  _checkProjectionIsValid (fact) {
+    // TODO current solution only requires last entry in context array so maybe this should just be a single property
```

Comment:
Was this always the case? The reason to have context be an array is to handle nested created facts, right?

Reply:
I don't know why we ever needed that. this._getFactChecker().getCreatingActs(fact, ssid, context) can get all acts where fact was created. And the 'should project a series of CREATE facts' still passes.

Reply:
Unless i'm missing something i don't see why we need it

Reply:
Okay, so I don't know if it's necessary, but it enables stuff like (pseudo-model)
```
[car accident]: CREATE('[causing car]', ['victims car]'), 
[causing car]: CREATE([driver], [license plate], [passengers]), 
[victims car]: CREATE([driver], [license plate], [passengers]), 
[driver]: CREATE([license])
[passenger]: CREATE([license])
```

Then if you want to get the license of the driver that was in the causing car, you can do a projection with context `["[causing car]", "[driver]"]`, does that make sense?

Reply:
I see i added a test for this and it does indeed fail good looking out :+1: